awscli/customizations/s3/s3handler.py

```diff
@@ -299,6 +310,8 @@ def _enqueue_multipart_upload_tasks(self, filename,
         # First we need to create a CreateMultipartUpload task,
         # then create UploadTask objects for each of the parts.
         # And finally enqueue a CompleteMultipartUploadTask.
+        if not self._is_upload_chunksize_valid(filename):
```

Comment:
I think this may mess up the progress bar because it is still being counted as a file to upload/copy even though it is being skipped with a part size of zero. It even looks like it in the updated output where it says that 2 of 2 parts is completed with 1 file remaining. I would recommend putting the logic in an if statement somewhere here: https://github.com/aws/aws-cli/blob/develop/awscli/customizations/s3/s3handler.py#L191. Mainly before the task gets counted here: https://github.com/aws/aws-cli/blob/develop/awscli/customizations/s3/s3handler.py#L220. It would also save you from having to duplicate the logic in both the copy and upload multipart enqueuing methods.